src/content/en/tools/chrome-devtools/javascript/overview.md

```diff
@@ -0,0 +1,140 @@
+
+* A line of code isn't producing the value you expected. If that value
+  is used as input to other parts of your code, then those other parts
+  are going to produce incorrect results, too.
+* You expect your script to execute in one order, but it's actually executing
+  in some other order.
+
+When you normally interact with a page, you're interacting with the end
+result of your script's execution. To debug the two mistakes above, it would
+be helpful to be able to view the intermediary values of functions and
+variables to determine if they are producing the expected results, and to
+step through a script's execution line-by-line to make sure that the script
+is executing in the expected order. This is exactly what debugging with
+DevTools enables.
+
+## The debugging workflow
+
+Below is the general workflow that you should follow when debugging your app.
+
+### Step 1: Reproduce the bug
+
+Find a way to consistently reproduce the bug. If somebody else reported the
+bug and you can't consistently reproduce it, get more information until you can.
+
+Once you can reproduce it, eliminate any steps that are not relevant to
+reproducing the bug. For example, if the bug report says "scroll up and down
+five times and then click this button", but you can reproduce the bug by just
+clicking the button without scrolling, then scrolling is not needed to
+reproduce the bug. Eliminating unnecessary steps like this helps you find
+and fix your bugs faster.
+
+### Step 2: Set and trigger a breakpoint
+
+Once you can reproduce the bug, it's time to start debugging with
+DevTools by setting a breakpoint.
+
+**Breakpoints** are intentional stopping places in your code.
+When the browser encounters a breakpoint, it pauses the script on
+that exact line of code.
+
+You set breakpoints in DevTools. DevTools provides many different types
+of breakpoints, but they all fall into one of two general categories:
+
+* **Line-of-code breakpoints**. Set on a specific line of code. Useful
+  when you want to inspect a particular section of code.
+* **Event-based breakpoints**. Set to trigger when a certain type of general
+  event occurs, like a `click` event or an uncaught exception. When the
+  event occurs, DevTools automatically takes you to the code associated
+  with the event. Event-based breakpoints are useful when you don't know
+  what code is causing a bug. They are often much faster to set than
+  line of code breakpoints, too.
+
+Ideally, you want to set your breakpoint somewhere close to the code that's
+causing the bug, so that you don't need to step through a bunch of code
+that's not related to your bug. The breakpoint needs to be set on a line of
+code before the bug, so that you
+can inspect what's happening in your code leading up to the bug. However,
+since you don't know what code is causing the bug, setting breakpoints is
+somewhat of a guessing game. It gets easier with practice.
+
```

Comment:
When I don't have a good guess, I set random breakpoints at smaller and smaller halves until I narrow it down. Let me know if you want a complete explanation.